tests/Integration/Behaviour/Features/Context/Domain/ProductFeatureContext.php

```diff
@@ -55,6 +58,53 @@ public function setProductTaxRulesGroup(string $taxRulesGroupReference, string $
         Cache::clean('product_id_tax_rules_group_*');
     }
 
+    /**
+     * @Then The category of product :productName has reduction of :reductionPercent% for the customer :customerReference
+     */
+    public function addReductionOnProductCategoryForCustomerGroup(string $productName, float $reductionPercent, string $customerReference)
+    {
+        $productId = $this->getProductIdByName($productName);
+
+        $product = new Product($productId);
+        $category = new Category($product->id_category_default);
+        if (!$category) {
```

Comment:
Wrong way : `$category` will be an object !

Reply:
Fixed : I use Category::categoryExists instead

Reply:
And I did the same fix for customer existence check